tethys_apps/base/app_base.py

```diff
@@ -41,8 +45,59 @@ class TethysBase(TethysBaseMixin):
 
     def __init__(self):
         self._url_patterns = None
+        self._handler_patterns = None
         self._namespace = None
 
+    @staticmethod
+    def _resolve_ref_function(is_extension, ref, ref_type):
```

Comment:
@msouff please add docstrings to the new methods: `_resolve_ref_function` and `_resolve_bokeh_handler` using the Google style. Also, let's move `is_extension` to the back of the argument list.